### DISCUSSION TITLE
(From Nx) Please rename all nrwl references to nx

### DISCUSSION THREAD

**Discussion by JamesHenry** [renovatebot/renovate#32938] [Suggest an Idea]
### Tell us more.

Hi Folks!

James from Nx here - please could you rename all the references to "nrwlMonorepo/nrwl" to be nx instead?

We do not use the nrwl name anywhere in npm packages any more and we'd love to avoid confusion for new users when renovate opens a "nrwl monorepo" PR.

We'd love to rename our GitHub org to Nx if we could, but sadly that's not an option at the moment.

Many thanks,

James

**Comment by rarkins** [comment]
You're welcome to submit a PR for the change. I assume that you mean that any packages with source as `https://github.com/nrwl/*` should be identified as "nx monorepo" instead of "nrwl monrepo"?

Is identifying them by source repo the best approach or are they all scoped now and could be identified by `@nx/` npm scope instead?

**Reply by JamesHenry** [reply]
I was hoping to before opening this, but a quick search did not reveal anything so I thought maybe the definitions don't live in this repo?

![image](https://github.com/user-attachments/assets/61bf5349-1764-445a-9870-2003ac77ea0d)

Yes, the `@nrwl/` scoped packages are no longer published to npm at all. There is the `nx`, `nx-cloud`, and `@nx/*` packages, that's it. The `nx` and `@nx/*` packages are all published from the same repo: https://github.com/nrwl/nx, the `nx-cloud` package is closed source.

**Reply by rarkins** [reply]
The line you found is the only place where it's defined. All the grouping is done automatically based on that source.

But let's back up a step: is it mandatory to group all updates from packages published from the github.com/nrwl together? Or are there some which are versioned and updated independently of each other? The monorepo grouping like this in Renovate is usually meant for packages which usually have to be updated together in lockstep.

**Reply by JamesHenry** [reply]
Yes, all the packages from that repo are versioned in lockstep. The closed source `nx-cloud` package is the only exception.

To give you a concrete example, this PR should say `chore: update nx monorepo to v20.2.0`, not `chore: update nrwl monorepo to v20.2.0`:

https://github.com/angular-eslint/angular-eslint/pull/2162

In which case I can open a PR to change the key name to `"nx"` there and it will just work?

**Reply by JamesHenry** [reply]
Will the website be auto generated from that source too? 

https://docs.renovatebot.com/presets-group/#groupnrwlmonorepo

**Reply by rarkins** [reply]
You're saying "repo" and not "org" so does that mean we only need to group packages from github.com/nwrl/nx? None of https://github.com/orgs/nrwl/repositories?type=source need to be grouped together/updated in lockstep?

**Reply by JamesHenry** [reply]
Any additional packages coming from other repos under the org won’t have the exact same versions, their releases are not strictly coordinated or anything.

If you’re asking whether conceptually a bump to nx-cloud (which comes from a private repo under the org)  could be combined in the same PR as the packages from the nx repo the answer is definitely yes, everything we publish is related.

But it also wouldn’t be an issue if there were separate PRs for things from within the nx repo vs outside, whichever you think is most straightforward on your side.

**Reply by rarkins** [reply]
So in that case it sounds like we should remove this nwrl rule from group:monorepo and instead have a more regular "nx monorepo" rule for https://github.com/nwrl/nx. To maintain backwards compatibility we have a migration from nwrl to nx too

**Reply by JamesHenry** [reply]
E.g. You can see nx-cloud has version `19.1.0`: https://www.npmjs.com/package/nx-cloud (it will rarely need to change)

It is not aligned with the [`nx` + `@nx/*`] packages which all have version `20.2.0`.

But if we cut a release of either group, it would be reasonable to reevalute the latest version of the other in a user's repo.

It's possible that is a misinterpretation of what you are asking, but hopefully the example makes it a bit clearer where we're coming from

**Reply by JamesHenry** [reply]
@rarkins That sounds a bit more involved, is that something you/your team will be handling?

**Reply by rarkins** [reply]
@RahulGautamSingh can you make the above change?
- Remove nwrl definition
- Add github.com/nwrl/nx monorepo definition as "nx"
- Add migration for group: and package: presets so we don't break anyone

**Reply by JamesHenry** [reply]
Thank you in advance @RahulGautamSingh!

**Reply by JamesHenry** [reply]
Should this be converted to an issue to make it easier to track?

**Reply by rarkins** [reply]
https://github.com/renovatebot/renovate/issues/32941

**Reply by RahulGautamSingh** [reply]
Fix: https://github.com/renovatebot/renovate/pull/33294
Sorry for the delay 🙌

**Reply by JamesHenry** [reply]
Thanks folks!

**Reply by JamesHenry** [reply]
Looking good in the wild, thanks again!

https://github.com/angular-eslint/angular-eslint/pull/2191
